### PR TITLE
feat: Add permission functions to ts-sdk

### PR DESCRIPTION
**Pull request by Juan-LukeKlopper**
this pull request introduces support for a series of new state transitions pertaining to the newly implemented permissions module. The following message types have been meticulously added:

- MsgClaimVoucher
- MsgCreateNamespace
- MsgDeleteNamespace
- MsgRevokeNamespaceRoles
- MsgUpdateNamespace
- MsgUpdateNamespaceRoles
- MsgUpdateParams

These state transitions are integral to the effective management and evolution of namespaces within the Injective ecosystem. They provide the necessary capabilities for creating, updating, deleting namespaces, and handling roles and permissions with precision.

Accompanying each of these additions are corresponding spec files which ensures that the new state transitions are robustly supported and are fully testable, adhering to high standards of quality and maintaining the integrity of the codebase.

## Summary by CodeRabbit

- **New Features**
	- Introduced a new API for interacting with permissions on the Injective blockchain, enabling users to fetch addresses by role, address roles, and all namespaces.
	- Added functionality to retrieve module parameters and vouchers associated with specific addresses.
	- Implemented a transformer for processing permissions-related data, enhancing usability of gRPC responses.
- **Bug Fixes**
	- Improved error handling for gRPC requests to provide better context in case of failures.
- **Tests**
	- Added unit tests for new API methods and transformers to ensure functionality and reliability.

### CODE REVIEW

**Comment by Juan-LukeKlopper** [comment]
This pull request originally introduced support for a series of new state transitions pertaining to the newly implemented permissions module. The following message types were added:

- **MsgClaimVoucher**
- **MsgCreateNamespace**
- **MsgDeleteNamespace**
- **MsgRevokeNamespaceRoles**
- **MsgUpdateNamespace**
- **MsgUpdateNamespaceRoles**
- **MsgUpdateParams**

These state transitions are essential for the effective management and evolution of namespaces within the Injective ecosystem, providing the necessary functionality for creating, updating, deleting namespaces, and handling roles and permissions with precision.

**This PR has now been updated to include additional queries** for retrieving data related to the permissions module. The newly added queries are:

- **fetchModuleParams**
- **fetchAllNamespaces**
- **fetchNamespaceByDenom**
- **fetchAddressesByRole**
- **fetchAddressRoles**
- **fetchVouchersForAddress**

As with the state transitions, accompanying spec files have been added to ensure that the new queries are thoroughly tested, maintaining high standards of code quality and ensuring the integrity of the codebase.

File: packages/sdk-ts/src/client/chain/transformers/ChainGrpcPermissionsTransformer.ts [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+            sendsPaused: grpcNamespace.sendsPaused,
+            burnsPaused: grpcNamespace.burnsPaused,
+            rolePermissions: grpcNamespace.rolePermissions,
+            addressRoles: grpcNamespace.addressRoles,
+        }
+    }
+
+    static addressRolesResponseToAddressRoles(
+        response: InjectivePermissionsV1Beta1Query.QueryAddressRolesResponse,
+    ){ return response.roles.map((role) => {
```
Comment (ThomasRalee) [inline packages/sdk-ts/src/client/chain/transformers/ChainGrpcPermissionsTransformer.ts:38]: line break here

File: packages/sdk-ts/src/client/chain/transformers/ChainGrpcPermissionsTransformer.ts [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+            return {
+              roles: role,
+          }
+        })
+    }
+
+    static addressesByRolesResponseToAddressesByRoles (
+        response: InjectivePermissionsV1Beta1Query.QueryAddressesByRoleResponse,
+    ){ return response.addresses.map((address) => {
```
Comment (ThomasRalee) [inline packages/sdk-ts/src/client/chain/transformers/ChainGrpcPermissionsTransformer.ts:47]: line break here

File: packages/sdk-ts/src/client/chain/grpc/ChainGrpcPermissionsApi.ts [hunk 3]
```diff
@@ -0,0 +1,219 @@
...
+      })
+    }
+  }
+
+  async fetchNamespaceByDenom({
+    denom,
+    include_roles,
+  }: {
+    denom: string
+    include_roles: boolean
```
Comment (ThomasRalee) [inline packages/sdk-ts/src/client/chain/grpc/ChainGrpcPermissionsApi.ts:156]: lets use camelCase here

**Comment by Juan-LukeKlopper** [comment]
I’ve updated the PR with line breaks and variable name change per the review comments!

**Review by ThomasRalee — APPROVED** [review]

**Pull request merged**